### ISSUE TITLE
Can setInterval and network I/O be moved out of LDClient#init?

### ISSUE THREAD

**Issue by ttahmouch** [launchdarkly/node-client#8]
We currently have a problem where we are not able to execute LDClient#init in a child process and expect the process to return an exit code. This is perceivably because the process may never exit its own event loop because setInterval will keep the event loop busy forever.

Is there a reason that instantiating the client in memory needs to begin [flushing every 5 seconds](https://github.com/launchdarkly/node-client/blob/master/index.js#L150)? Could this perhaps be a lazy and start when LDClient#toggle is invoked?

**Comment by jkodumal** [comment]
Would calling [unref](https://nodejs.org/api/timers.html#timers_unref) on our timer be sufficient? The `flush` call does not hit the network if there are no queued events, so simply calling `init` should not do any network IO.

**Comment by ttahmouch** [comment]
I just tested these in separate node processes. I never knew `unref` existed. I think it would solve the problem.

```javascript
// This never exited.
setInterval(function() {console.log('test')}, 1000)
```

```javascript
// This exited immediately.
setInterval(function() {console.log('test')}, 1000).unref()
```

```javascript
// This never exited because the server existed on the event loop alongside the timer.
var http = require('http');
http.createServer(function (req, res) {
  res.writeHead(200, {'Content-Type': 'text/plain'});
  res.end('Hello World\n');
}).listen(1337, '127.0.0.1');
console.log('Server running at http://127.0.0.1:1337/');

setInterval(function() {console.log('test')}, 1000).unref()
```

**Comment by jkodumal** [comment]
Released in v 0.2.1.
